RepositoryCompiler/CodeModel/CodeParsers/CSharp/CSharpMetricCalculator.cs

```diff
@@ -35,6 +37,60 @@ public CaDETClassMetrics CalculateClassMetrics(CaDETClass parsedClass)
             return Math.Round(1 - methodFieldAccess/maxCohesion, 3);
         }
 
+        private double? GetTightOfCohesion(CaDETClass parsedClass)
+        {
+            int N = GetNumberOfMethodsDeclared(parsedClass);
+            double NP = (N * (N - 1)) / 2;
+            if (NP == 0) return null;
+
+            return NumberOfDirectConnections(parsedClass) / NP;
+        }
+
+        private int NumberOfDirectConnections(CaDETClass parsedClass)
+        {
+            int mapIndex = 1;
+            var map = new Dictionary<int, List<CaDETMember>>();
+
+            foreach (var method in parsedClass.Methods)
+            {
+                map.Add(mapIndex, OwnAccessedFieldsAndAccessors(method));
+                mapIndex++;
+            }
+
+            int countDirectConnections = 0;
+
+            foreach (var pair1 in map)
+            {
+                List<CaDETMember> ownAccessedFieldsAndAccessors1 = pair1.Value;
+                foreach (var pair2 in map)
+                {
+                    List<CaDETMember> ownAccessedFieldsAndAccessors2 = pair2.Value;
```

Comment:
A tip for naming things:
1) Describe the thing in a sentence or two (easier to do in Serbian).
2) Remove any words that are not crucial to convey the meaning.
3) Remove any words that can be inferred from the Type, containing function, containing class, or containing module.
4) See if what remains sufficiently describes the thing.
5) Rename.